guides/source/active_record_callbacks.md

```diff
@@ -329,124 +763,267 @@ Just as with validations, it is also possible to skip callbacks by using the fol
+user, allowing us to create the user without sending the welcome email.
+
+WARNING: Using the Active Record Suppressor, while potentially beneficial for
+selectively controlling callback execution, can introduce complexity and
+unexpected behavior. Suppressing callbacks can obscure the intended flow of your
+application, leading to difficulties in understanding and maintaining the
+codebase over time. Carefully consider the implications of suppressing
+callbacks, ensuring thorough documentation and thoughtful testing to mitigate
+risks of unintended side effects, performance issues, and test failures.
+
+Halting Execution
+-----------------
+
+As you start registering new callbacks for your models, they will be queued for
+execution. This queue will include all of your model's validations, the
+registered callbacks, and the database operation to be executed.
+
+The whole callback chain is wrapped in a transaction. If any callback raises an
+exception, the execution chain gets halted and a **rollback** is issued, and the
+error will be re-raised.
+
+```ruby
+class Product < ActiveRecord::Base
+  before_validation do
+    raise "Price can't be negative" if total_price < 0
   end
 end
+
+Product.create # raises "Price can't be negative"
 ```
 
-```irb
-irb> user = User.first
-=> #<User id: 1>
-irb> user.articles.create!
-=> #<Article id: 1, user_id: 1>
-irb> user.destroy
-Article destroyed
-=> #<User id: 1>
+This unexpectedly breaks code that does not expect methods like `create` and
+`save` to raise exceptions.
+
+Instead, you should use `throw :abort` to intentionally halt the chain. If any
+callback throws `:abort`, the process will be aborted and `create` will return
+false.
+
+```ruby
+class Product < ActiveRecord::Base
+  before_validation do
+    throw :abort if total_price < 0
+  end
+end
+
+Product.create # => false
 ```
 
+WARNING: If an exception occurs during the callback chain, Rails will re-raise
+it unless it is an `ActiveRecord::Rollback`, `ActiveRecord::RecordNotSaved` or
+`ActiveRecord::RecordInvalid` exception.
+
+
+When `throw :abort` is called in any destroy callback, `destroy` will return
+false:
+
+```ruby
+class User < ActiveRecord::Base
+  before_destroy do
+    throw :abort if still_active?
+  end
+end
+
+User.first.destroy # => false
+```
+
+However, it will raise an `ActiveRecord::RecordNotDestroyed` when calling
+`destroy!`.
+
+```ruby
+User.first.destroy! # => raises an ActiveRecord::RecordNotDestroyed
+```
+
+In addition to the behaviors mentioned, it's important to note that when `throw
+:abort` is called in a `before_* callback` (such as `before_save`,
+`before_create`, or `before_update`), it will raise an
+`ActiveRecord::RecordNotSaved` exception. This exception indicates that the
+record was not saved due to the callback's interruption. Therefore, when using
+`throw :abort` in `before_*` callbacks, you should be prepared to handle the
+`ActiveRecord::RecordNotSaved` exception.
```

Comment:
With the above mentioned changes, that make clear that ActiveRecord::RecordNotSaved` exception can be raised, I think we can remove this paragraph altogether.
```suggestion
```

Reply:
✅